xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -445,7 +446,7 @@ public Result selectConfig(PickSubchannelArgs args) {
         }
       }
       RetryPolicy retryPolicy =
-          selectedRoute == null ? null : selectedRoute.routeAction().retryPolicy();
+          selectedRoute == null ? null : selectedRoute.routeAction.retryPolicy();
```

Comment:
Not directly related to your chane, but `selectedRoute == null` checks are redundant. (also line 438)

```suggestion
          selectedRoute.routeAction.retryPolicy();
```

Reply:
I didn't notice that, and it is pretty misleading. But it's also not alone. I'm fixing this in https://github.com/grpc/grpc-java/pull/11867